meson.build

```diff
@@ -73,20 +90,9 @@ warning_c_flags = [
 ]
 
 # Setup warning flags for c and cpp
-foreach extra_arg : warning_flags
-  if cc.has_argument (extra_arg)
-    add_project_arguments([extra_arg], language: 'c')
-  endif
-  if cxx.has_argument (extra_arg)
-    add_project_arguments([extra_arg], language: 'cpp')
-  endif
-endforeach
-
-foreach extra_arg : warning_c_flags
-  if cc.has_argument (extra_arg)
-    add_project_arguments([extra_arg], language: 'c')
-  endif
-endforeach
+add_project_arguments(cc.get_supported_arguments(warning_c_flags), language: 'c')
+add_project_arguments(cc.get_supported_arguments(warning_flags), language: 'c')
+add_project_arguments(cxx.get_supported_arguments(warning_flags), language: 'cpp')
```

Comment:
Your code is very efficient and concise. First, instead of using a foreach loop to set warning flags, you can use the cc.get_supported_arguments() and cxx.get_supported_arguments() functions to get all the supported warning flags for that language at once. This improves the readability and maintainability of your code.
Also, by specifying the appropriate language for each project through the language: parameter, we ensure that the warning flags are applied correctly in both C and C++ projects. This helps maintain consistency in our projects and prevents potential errors.
So, your code change is quite effective and it can help we build a better programming environment.